src/nnet/nnet-parametric-relu.h

```diff
@@ -0,0 +1,266 @@
+// nnet/nnet-parametric-relu.h
+
+// Copyright 2016 Brno University of Technology (author: Murali Karthick B)
+//           2011-2014  Brno University of Technology (author: Karel Vesely)
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+
+#ifndef KALDI_NNET_NNET_PARAMETRIC_RELU_H_
+#define KALDI_NNET_NNET_PARAMETRIC_RELU_H_
+
+#include <string>
+
+#include "nnet/nnet-component.h"
+#include "nnet/nnet-utils.h"
+#include "cudamatrix/cu-math.h"
+
+namespace kaldi {
+namespace nnet1 {
+
+class ParametricRelu : public UpdatableComponent {
+ public:
+  ParametricRelu(int32 dim_in, int32 dim_out):
+    UpdatableComponent(dim_in, dim_out),
+    alpha_(dim_out), 
+    beta_(dim_out),
+    alpha_corr_(dim_out),
+    beta_corr_(dim_out),
+    clip_gradient_(0.0)
+  { }
+  ~ParametricRelu()
+  { }
+
+  Component* Copy() const { return new ParametricRelu(*this); }
+  ComponentType GetType() const { return kParametricRelu; }
+
+  void InitData(std::istream &is) {
+    // define options
+    float init_alpha = 1, init_beta = 0.25;
+    float fix_alpha = 0.0, fix_beta = 0.0;
+
+    // parse config
+    std::string token;
+    while (is >> std::ws, !is.eof()) {
+      ReadToken(is, false, &token);
+      /**/ if (token == "<InitAlpha>")  ReadBasicType(is, false, &init_alpha);
+      else if (token == "<InitBeta>")  ReadBasicType(is, false, &init_beta);
+      else if (token == "<BiasLearnRateCoef>")  ReadBasicType(is, false, &bias_learn_rate_coef_);
```

Comment:
Is that really how you are doing it?  Given that a isn't a bias parameter, that doesn't seem appropriate.
Repurposing existing variables for different purposes for which their name is not appropriate, is not something that I've seen before in Kaldi code.